server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java

```diff
@@ -329,6 +335,23 @@ public void write(TimeValuePair pair) {
         IoTDB.metaManager.updateLastCache(node, pair, false, Long.MIN_VALUE);
       }
     }
+
+    public boolean checkEmptyContainer() {
+      if (node == null) {
+        return false;
+      } else {
+        return node.getLastCacheContainer().isEmptyContainer();
+      }
+    }
+
+    public void markEmpty() {
+      if (node == null) {
+        return;
+      } else {
+        node.setLastCacheContainer(EmptyLastCacheContainer.getInstance());
```

Comment:
We can't directly use EmptyLastCacheContainer to set the node's LastCacheContainer because it may be already set to normal LastCacheContainer through another insert thread. If so, this operation may cause wrong cache container.

Reply:
Since the method `markEmpty` is called in the method `calculateLastPairForSeriesLocally` when the system restarts and the `LastCacheContainer` is initialized as null. So I think it may be safe to use `EmptyLastCacheContainer` to set the node's `LastCacheContainer`.